augur/filter.py

```diff
@@ -788,10 +788,18 @@ def apply_filters(metadata, exclude_by, include_by):
     for filter_function, filter_kwargs in exclude_by:
         # Apply the current function with its given arguments. Each function
         # returns a set of strains that passed the corresponding filter.
-        passed = metadata.pipe(
-            filter_function,
-            **filter_kwargs,
-        )
+        try:
+            passed = metadata.pipe(
+                filter_function,
+                **filter_kwargs,
+            )
+        except Exception as e:
+            if filter_function.__name__ == 'filter_by_query':
+                if type(e) == pd.core.computation.ops.UndefinedVariableError:
```

Comment:
Generally comparison of classes is conventionally done with `is` not `==` since they're identical values, but also instance type checking is often done with `isinstance()` instead of `type()` to allow for subclasses.

```suggestion
                if isinstance(e, pd.core.computation.ops.UndefinedVariableError):
```